Update Graph Class, move /graph_search from planners to utils.

Srujan-D opened the pull request:
1) Add method in Graph class (gennav.utils.graph) to delete edge connecting given two nodes from the graph.
2) Move /graph_search from gennav.planners to gennav.utils.

Srujan-D committed:
Added Delete Edge method in Graph Class.

Srujan-D committed:
Moved graph_search (init) to utils.

Srujan-D committed:
Moved graph_search (astar) form planners to utils.

Srujan-D committed:
Update __init__.py

Srujan-D committed:
Updated import line for astar.

Srujan-D committed:
Update import line for astar.

Srujan-D committed:
Update import line for astar.

threewisemonkeys-as commented on gennav/utils/graph.py:
```diff
@@ -33,6 +33,32 @@ def add_edge(
+            to_node (gennav.utils.common.Node): ending node of the edge.
+        """
+
+        if len(self.edges[from_node]):
```
Can you make this if .. == 0:

Makes it more explicit

Srujan-D replied:
Done.

threewisemonkeys-as commented on gennav/utils/graph.py:
```diff
@@ -33,6 +33,32 @@ def add_edge(
+
+            del self.distances[(from_node, to_node)]
+        else:
+            print("Edge does not exist")
```
You could raise ValueError here

Srujan-D replied:
Done.

Srujan-D committed:
Update graph.py

threewisemonkeys-as commented:
Could you look at the pytest errors?

Srujan-D committed:
Update astar.py to suit the modified Node class.

Srujan-D commented:
> Could you look at the pytest errors?

Done, they were about updating the astar code to suit the modified Node class.

threewisemonkeys-as commented on gennav/utils/graph.py:
```diff
@@ -33,6 +33,32 @@ def add_edge(
+            del self.distances[(from_node, to_node)]
+
+        if len(self.edges[to_node]) == 0:
+            raise ValueError
```
Add string describing what error is

Srujan-D replied:
Done.

Srujan-D committed:
Update graph.py

1) Changed argument type from Node to RobotState.
2) Modified other methods and doc-strings accordingly.
3) Deleted unnecessary self.dist attribute.

threewisemonkeys-as commented:
Should I merge this or are you going to make the changes to PRM on this PR itself?

Srujan-D commented:
> Should I merge this or are you going to make the changes to PRM on this PR itself?

If you want different PR for those changes, you can merge this PR.
I have done the necessary changes to the concerned codes, but the visualize graph isn't working at all. The builds are passing whenever I don't include the visualization lines in the test file. I need some time to figure out what's causing the fail.

threewisemonkeys-as commented:
We can leave the visualisation for later. Lets get the core functionality done. I'll merge this. You can open another PR for the other changes.
